### PR TITLE
Add more documentation for @wordpress/env

### PR DESCRIPTION
**Pull request by noisysocks**
Quick attempt at fleshing out some of our documentation for the `@wordpress/env` tool. These are the docs that are shown here:

https://developer.wordpress.org/block-editor/packages/packages-env/

When https://github.com/WordPress/gutenberg/pull/18986 is merged, we can also update `docs/contributors/getting-started.md` to instruct users to use `wp-env` instead of `npm run env`.

cc. @epiqueras @mapk

### CODE REVIEW

File: packages/env/README.md [hunk 1]
```diff
@@ -1,37 +1,141 @@
...
+## Quick (tl;dr) instructions
+
+Ensure that Docker is running, then:
+
+```sh
+$ cd /path/to/a/wordpress/plugin
+$ npx wp-env start
+```
+
+The local environment will be available at http://localhost:8000.
```
Comment (epiqueras) [inline packages/env/README.md]:
```suggestion
The local environment will be available at http://localhost:8888.
```

File: packages/env/README.md [hunk 2]
```diff
@@ -1,37 +1,141 @@
...
+
+## Instructions
+
+### Installation
+
+`wp-env` requires Docker to be installed. There are instructions available for installing Docker on [Windows 10 Pro](https://docs.docker.com/docker-for-windows/install/), [all other version of Windows](https://docs.docker.com/toolbox/toolbox_install_windows/), [macOS](https://docs.docker.com/docker-for-mac/install/), and [Linux](https://docs.docker.com/v17.12/install/linux/docker-ce/ubuntu/#install-using-the-convenience-script).
```
Comment (epiqueras) [inline packages/env/README.md]:
```suggestion
`wp-env` requires Docker to be installed. There are instructions available for installing Docker on [Windows 10 Pro](https://docs.docker.com/docker-for-windows/install/), [all other versions of Windows](https://docs.docker.com/toolbox/toolbox_install_windows/), [macOS](https://docs.docker.com/docker-for-mac/install/), and [Linux](https://docs.docker.com/v17.12/install/linux/docker-ce/ubuntu/#install-using-the-convenience-script).
```

File: packages/env/README.md [hunk 3]
```diff
@@ -1,37 +1,141 @@
...
 ```
 
-### `$ wp-env start --help`
+Then, start the local environment:
+
+```sh
+$ wp-env start
+```
+
+Finally, navigate to http://localhost:8000 in your web browser to see WordPress running with the local WordPress plugin or theme running and activated.
```
Comment (epiqueras) [inline packages/env/README.md]:
```suggestion
Finally, navigate to http://localhost:8888 in your web browser to see WordPress running with the local WordPress plugin or theme running and activated.
```

File: packages/env/README.md [hunk 4]
```diff
@@ -1,37 +1,141 @@
...
+
+### 1. Check that `wp-env` is running
+
+First, check that `wp-env` is running. One way to do this is to have Docker print a table with the currently running containers:
+
+```sh
+$ docker ps
+```
+
+In this table, by default, you should see three entries: `wordpress` with port 8888, `wordpress` with port 8889 and `mariadb` with port 3306.
```
Comment (epiqueras) [inline packages/env/README.md]:
```suggestion
In this table, by default, you should see three entries: `wordpress` with port 8888, `tests-wordpress` with port 8889 and `mariadb` with port 3306.
```

**Review by epiqueras — CHANGES_REQUESTED** [review]

File: packages/env/README.md [hunk 5]
```diff
@@ -92,3 +194,5 @@ Positionals:
 The file should be located in the same directory from which you run `wp-env` commands for a project. So if you are running `wp-env` in the root directory of a plugin, `wp-env.json` should also be located there. 
 
 Each item in the `themes` or `plugins` array should be an absolute or relative path to the root of a different theme or plugin directory. Relative paths will be resolved from the current working directory, which means they will be resolved from the location of the `wp-env.json` file.
+
+<br/><br/><p align="center"><img src="https://s.w.org/style/images/codeispoetry.png?1" alt="Code is Poetry." /></p>
```
Comment (ZebulanStanphill) [inline packages/env/README.md]: Why is this here? Also, the `align` attribute is not valid HTML5.

**Review by ZebulanStanphill — CHANGES_REQUESTED** [review]

File: packages/env/README.md (hunk 5 above)
Comment (aduth) [reply packages/env/README.md]: > Why is this here? Also, the `align` attribute is not valid HTML5.

These have been included in all of the package READMEs, so I assume it was added for consistency's sake.

Whether it makes sense to include this image... I don't see a ton of value in it it, but I grant it's likely intended more as personality / "flair" to associate with WordPress.

Re: `align`, I think it's out of necessity for it to appear centered in GitHub. I've seen (and used) it in Markdown files in other projects. I don't know that there's an alternative that works equally well for GitHub Markdown file previews.

**Comment by gziolo** [comment]
> When #18986 is merged, we can also update docs/contributors/getting-started.md to instruct users to use wp-env instead of npm run env.

What’s wrong with having npm run env command in scripts section which acts like a documentation for node users?

File: packages/env/README.md (hunk 4 above)
Comment (noisysocks) [reply packages/env/README.md]: It shows in `docker ps` as `wordpress` but agree this is clearer.

**Review by epiqueras — APPROVED** [review]

File: packages/env/README.md [hunk 6]
```diff
@@ -1,38 +1,141 @@
...
+`wp-env` lets you easily set up a local WordPress environment for building and testing plugins and themes. It's simple to install and requires no configuration.
 
-## Usage
+## Quick (tl;dr) instructions
+
+Ensure that Docker is running, then:
+
+```sh
+$ cd /path/to/a/wordpress/plugin
+$ npx wp-env start
```
Comment (ZebulanStanphill) [inline packages/env/README.md]: Here `npx wp-env` is used, but everywhere else it's just `wp-env`. Perhaps a note explaining the difference (local vs global installation) should be included?

**Review by ZebulanStanphill — APPROVED** [review]
One minor comment, but everything else looks way more clear and helpful than what's currently in `master`. :+1:

File: packages/env/README.md (hunk 6 above)
Comment (noisysocks) [reply packages/env/README.md]: This section is intended to be an overly brief "tl;dr" section aimed at experienced developers. With that in mind I'd prefer to leave looking up what `npx` is as an exercise for the reader.

File: packages/env/README.md [hunk 7]
```diff
@@ -1,38 +1,141 @@
...
+
+When all else fails, you can try forcibly removing all of the underlying Docker containers and volumes, the underlying WordPress directory, and starting again from scratch.
+
+To nuke everything:
+
+**⚠️ WARNING: This will permanently delete any posts, pages, media, etc. in the local WordPress installation.**
+
+```sh
+$ docker rm -f $(docker ps -aq)
+$ docker volume rm -f $(docker volume ls -q)
```
Comment (mcsf) [inline packages/env/README.md]: > This will permanently delete any posts, pages, media, etc. **in the local WordPress installation**.

The copy should make it much clearer that these deletions will affect all possible Docker containers on the user's system, not just those concerning the immediate development environment.
